### PR TITLE
DBZ-227 Enabling Debezium to be built on Java 9

### PR DESCRIPTION
**Pull request by gunnarmorling**
https://issues.jboss.org/browse/DBZ-227

### CODE REVIEW

File: support/checkstyle/pom.xml [hunk 1]
```diff
@@ -20,6 +20,10 @@
         <project.build.sourceEncoding>UTF-8</project.build.sourceEncoding>
         <maven.compiler.source>1.8</maven.compiler.source>
         <maven.compiler.target>1.8</maven.compiler.target>
+
+        <version.enforcer.plugin>3.0.0-M1</version.enforcer.plugin>
```
Comment (gunnarmorling) [inline support/checkstyle/pom.xml]: That duplication isn't great, but it's required as this POM currently doesn't extend the parent one. We should change that eventually, there is an issue for it already.

File: pom.xml [hunk 2]
```diff
@@ -723,5 +726,25 @@
...
+            </activation>
+            <build>
+                <pluginManagement>
+                    <plugins>
+                        <plugin>
+                            <groupId>org.apache.maven.plugins</groupId>
+                            <artifactId>maven-surefire-plugin</artifactId>
+                            <!-- Connect is using javax.xml.bind.DatatypeConverter -->
+                            <configuration>
+                                <argLine>--add-modules java.xml.bind</argLine>
```
Comment (jpechane) [inline pom.xml]: This is the problem. You should not set `argLine` here as it is already ste for surefire locally. Please look at `debug.argline` property and use the same approach. Still I am not sure this is going to work as Kafka does not work with Java 9 and I see exceptions like
```
2017-10-19 15:50:45,863 ERROR KAF Error when freeing index buffer
java.lang.NoSuchMethodError: sun.nio.ch.DirectBuffer.cleaner()Lsun/misc/Cleaner;
	at kafka.log.AbstractIndex.forceUnmap(AbstractIndex.scala:206)
	at kafka.log.AbstractIndex$$anonfun$delete$1$$anonfun$apply$mcV$sp$1.apply$mcV$sp(AbstractIndex.scala:153)
	at kafka.utils.CoreUtils$.swallow(CoreUtils.scala:78)
	at kafka.utils.Logging$class.swallowWarn(Logging.scala:94)
	at kafka.utils.CoreUtils$.swallowWarn(CoreUtils.scala:48)
	at kafka.utils.Logging$class.swallow(Logging.scala:96)
	at kafka.utils.CoreUtils$.swallow(CoreUtils.scala:48)
	at kafka.log.AbstractIndex$$anonfun$delete$1.apply$mcV$sp(AbstractIndex.scala:153)
	at kafka.log.AbstractIndex$$anonfun$delete$1.apply(AbstractIndex.scala:148)
	at kafka.log.AbstractIndex$$anonfun$delete$1.apply(AbstractIndex.scala:148)
	at kafka.utils.CoreUtils$.inLock(CoreUtils.scala:213)
	at kafka.log.AbstractIndex.delete(AbstractIndex.scala:148)
	at kafka.log.LogSegment.delete(LogSegment.scala:501)
	at kafka.log.Log$$anonfun$delete$1.apply(Log.scala:1343)
	at kafka.log.Log$$anonfun$delete$1.apply(Log.scala:1343)
	at scala.collection.Iterator$class.foreach(Iterator.scala:891)
	at scala.collection.AbstractIterator.foreach(Iterator.scala:1334)
	at scala.collection.IterableLike$class.foreach(IterableLike.scala:72)
	at scala.collection.AbstractIterable.foreach(Iterable.scala:54)
	at kafka.log.Log.delete(Log.scala:1343)
	at java.base/java.lang.Iterable.forEach(Iterable.java:75)
	at io.debezium.kafka.KafkaServer.shutdown(KafkaServer.java:228)
	at io.debezium.kafka.KafkaCluster.shutdownReliably(KafkaCluster.java:355)
	at java.base/java.util.concurrent.ConcurrentHashMap$ValuesView.forEach(ConcurrentHashMap.java:4798)
	at io.debezium.kafka.KafkaCluster.shutdown(KafkaCluster.java:244)
	at io.debezium.relational.history.KafkaDatabaseHistoryTest.afterEach(KafkaDatabaseHistoryTest.java:70)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method)
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62)
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43)
	at java.base/java.lang.reflect.Method.invoke(Method.java:564)
	at org.junit.runners.model.FrameworkMethod$1.runReflectiveCall(FrameworkMethod.java:50)
	at org.junit.internal.runners.model.ReflectiveCallable.run(ReflectiveCallable.java:12)
	at org.junit.runners.model.FrameworkMethod.invokeExplosively(FrameworkMethod.java:47)
	at org.junit.internal.runners.statements.RunAfters.evaluate(RunAfters.java:33)
	at org.junit.runners.ParentRunner.runLeaf(ParentRunner.java:325)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:78)
	at org.junit.runners.BlockJUnit4ClassRunner.runChild(BlockJUnit4ClassRunner.java:57)
	at org.junit.runners.ParentRunner$3.run(ParentRunner.java:290)
	at org.junit.runners.ParentRunner$1.schedule(ParentRunner.java:71)
	at org.junit.runners.ParentRunner.runChildren(ParentRunner.java:288)
	at org.junit.runners.ParentRunner.access$000(ParentRunner.java:58)
	at org.junit.runners.ParentRunner$2.evaluate(ParentRunner.java:268)
	at org.junit.runners.ParentRunner.run(ParentRunner.java:363)
	at org.apache.maven.surefire.junit4.JUnit4Provider.execute(JUnit4Provider.java:283)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeWithRerun(JUnit4Provider.java:173)
	at org.apache.maven.surefire.junit4.JUnit4Provider.executeTestSet(JUnit4Provider.java:153)
	at org.apache.maven.surefire.junit4.JUnit4Provider.invoke(JUnit4Provider.java:128)
	at org.apache.maven.surefire.booter.ForkedBooter.invokeProviderInSameClassLoader(ForkedBooter.java:203)
	at org.apache.maven.surefire.booter.ForkedBooter.runSuitesInProcess(ForkedBooter.java:155)
	at org.apache.maven.surefire.booter.ForkedBooter.main(ForkedBooter.java:103)
```

Comment (rhauch) [reply pom.xml]: AK 1.0.0, when it comes out (hopefully soon), can be used with Java 9.

Comment (gunnarmorling) [reply pom.xml]: @jpechane so is this to say it's only not working when using the _debug_ profile? Tbh. I never use that, I just specify the standard options `-Dmaven.surefire.debug=true` or `-Dmaven.failsafe.debug=true`. Let's remove that profile perhaps?

> AK 1.0.0, when it comes out (hopefully soon), can be used with Java 9.

Ok, thanks for the heads-up, that's cool. I thought it's a quick fix here (and it works on my machine (TM) in the proposed form), but if it's actually more hazzle, then I'm happy to wait.

Comment (jpechane) [reply pom.xml]: I've prepared the commit with changes. It works for me now. Could you please verify that it is acceptable fro you too?

**Comment by gunnarmorling** [comment]
Thanks! Merging, @jpechane!

**Pull request merged**